src/lib/libpthread.js

```diff
@@ -1199,6 +1199,16 @@ var LibraryPThread = {
                         '_emscripten_check_mailbox',
                         '_emscripten_thread_mailbox_await'],
   $checkMailbox: () => {
+    // The mailbox is checked using a setTimeout() handler, but the application
+    // may have quit (or crashed) after the setTimeout() was registered,
+    // so first check if that might be the case, and no-op if so.
+#if EXIT_RUNTIME
+    if (runtimeExited) return;
+#endif
+#if !MINIMAL_RUNTIME
+    if (ABORT) return;
+#endif
+
     // Only check the mailbox if we have a live pthread runtime. We implement
     // pthread_self to return 0 if there is no live runtime.
     var pthread_ptr = _pthread_self();
```

Comment:
From the comment here it looks like the intention was that this check should take care of that case. 

Perhaps at least we don't need both of these checks?

CC @tlively who wrote this code originally and @brendandahl who recent fixed a shutdown race condition.

Reply:
`pthread_self()` is a native function, so it gets a boundary wrapper that aborts if it is attempted to be called without a live runtime.

Checking for `_pthread_self()` should be also needed in addition to checking for `ABORT`, since a given Worker might not host a live pthread at any given moment.

Reply:
It looks like there is a special callout here for `_pthread_self()`: https://github.com/emscripten-core/emscripten/blob/b132141e5c05df4ea9980fa7659b62f160c8ebaa/tools/emscripten.py#L900-L903

Perhaps that TODO can now be fixed and we can remove `pthread_self` and `_emscripten_proxy_execute_task_queue` those from `install_debug_wrapper`?

Reply:
That sounds like a separate PR to look into.

Reply:
One problem I see here is that we are kind of muddying the waters here about that it means to be a live runtime.   We have 3 different signals here: `pthread_self`, `ABORT` and `runtimeExited`.

I would love to have just a single condition that we could check here, but maybe we can just add a TODO to do further investigation.  For example, I would hope that on on abort or runtime exit then pthread_self would get set to null (since the thread is no longer hosting a pthread).   So I would *hope* these checks are redundant.  The fact that they are not redundant seems like a bug worth investigating.

So lgtm with a TODO to cleanup?

Reply:
Oh, wait, I just noticed that `callUserCallback` already contains these checks.

So maybe the issue is only with `-sMINIMAL_RUNTIME` where `callUserCallback` is doesn't contain the checks?

In that case I think maybe the correct fix here is to add the `ABORT` and `runtimeExited` checks to `callUserCallback` in libcore.js line 2117.

Reply:
(Also I guess the scope of the `callUserCallback` should be widened to be the whole body of `checkMailbox`)